services/explorer/src/pages/Faucet/Faucet.effects.ts

```diff
@@ -0,0 +1,105 @@
+import { useSnackbar } from 'notistack';
+import { useCallback, useEffect, useState } from 'react';
+import { faucetUri } from '../../config';
+import { useExplorerNetwork } from '../../hooks/useExplorerNetwork';
+import { useExplorerLoader } from '../../hooks/useLoader';
+
+export enum RequestLimit {
+  ONE = 1,
+  TWO = 2,
+  FIVE = 5,
+  SEVEN = 7,
+}
+export const useFaucetEffects = () => {
+  const { currentNetwork } = useExplorerNetwork();
+  const { isLoading, initLoading, finishLoading } = useExplorerLoader();
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const [address, setAddress] = useState<string>('');
```

Comment:
These should not be addres inside the state, because the only way to know when these are added inside the state is to use, useEffect which will not work as expected here. The address and amount should be handled by a form and on submit should call the request method from the API folder.